plugins/src/utils/src/lib/utils/find-in-file.ts

```diff
@@ -1,84 +1,64 @@
-import * as readline from 'node:readline';
-import * as fs from 'node:fs';
+import { Dirent } from 'node:fs';
+import * as fs from 'node:fs/promises';
 import * as path from 'node:path';
 
 /**
- * Reads a file line-by-line and checks if it contains the search pattern.
- * @param {string} filePath - The file to check.
+ * Searches for `.ts` files containing the search pattern.
+ * @param {string} baseDir - The directory to search. Should be absolute or resolved by the caller.
  * @param {RegExp | string} searchPattern - The pattern to match.
- * @returns {Promise<boolean>} - True if the pattern is found.
  */
-async function fileContainsPattern(
-  filePath: string,
-  searchPattern: string | RegExp
-): Promise<boolean> {
-  return new Promise((resolve, reject) => {
-    const stream = fs.createReadStream(filePath, { encoding: 'utf8' });
-    const rl = readline.createInterface({ input: stream });
+export async function findFilesWithPattern(baseDir: string, searchPattern: string) {
+  const resolvedBaseDir = path.resolve(baseDir);
 
-    let searchPatternFound = false;
+  const tsFiles: string[] = [];
+  for await (const file of findAllFiles(resolvedBaseDir, (file) => file.endsWith('.ts') && !file.endsWith('.spec.ts'))) {
```

Comment:
instead of hard-codeing the predicate, make it a option next to `searchPattern`.

```ts
findFilesWithPattern(baseDir: string, options: {searchPattern: Regex, filePredicate:() => ...}) {
//...
}
```